lib/config/fileconf.go

```diff
@@ -1553,11 +1568,18 @@ func (ip *InstallParams) Parse() (services.InstallerParams, error) {
 		SSHDConfig:      ip.SSHDConfig,
 	}
 
+	isV4OrGreater, err := ConfigVersionGTE(defaults.TeleportConfigVersionV4, version)
+	if err != nil {
+		return services.InstallerParams{}, trace.Wrap(err)
+	}
+	if isV4OrGreater {
+		install.InstallTeleport = false
+	}
```

Comment:
Maybe something like

```go
func (ip *InstallParams) setInstallTeleport(version string, install *services.InstallerParams) error {
	var err error

	// If install_teleport was set explicitly, use the given value.
	if ip.InstallTeleport != "" {
		install.InstallTeleport, err = apiutils.ParseBool(ip.InstallTeleport)
		return trace.Wrap(err)
	}

	// If install_teleport wasn't set explicitly, default to false for >= v4, true otherwise.
	install.InstallTeleport = true
	isV4OrGreater, err := configVersionGTE(defaults.TeleportConfigVersionV4, version)
	if err != nil {
		return trace.Wrap(err)
	}
	if isV4OrGreater {
		install.InstallTeleport = false
	}

	return nil
}

func (ip *InstallParams) Parse(version string) (services.InstallerParams, error) {
	install := services.InstallerParams{
		JoinMethod: ip.JoinParams.Method,
		JoinToken:  ip.JoinParams.TokenName,
		ScriptName: ip.ScriptName,
		SSHDConfig: ip.SSHDConfig,
	}

	ip.setInstallTeleport(version, &install)

	return install, nil
}
```

Needn't necessarily be it's own function, but the logic can be rewritten so that its easier to comprehend.